### DISCUSSION TITLE
undefined reference to `H5free_memory'

### DISCUSSION THREAD

**Discussion by buzhiqiuyuebai** [SeisSol/SeisSol#763] [Q&A] [answered]
I had no problems compiling the easi library before, but now I encountered this problem when compiling the easi library on a multi-node server. The previous four dependent packages did not have any problems during installation. Do you know why this problem occurs?
![image](https://user-images.githubusercontent.com/114474894/208243550-196b7ca6-8105-4866-979a-683042720ae1.png)

![image](https://user-images.githubusercontent.com/114474894/208242716-559d4b53-d758-4c4e-ac95-a2a74953ddc8.png)

**Comment by buzhiqiuyuebai** [comment]
I used an old version easi, it can be compiled successfully, maybe  the newer version easi has some probloms, but I am not sure.

**Comment by sebwolf-de** [comment, marked as answer]
It is very important that you use the same hdf5/netcdf version everywhere (asagi, easi, SeisSol). Also all libraries have to be compiled with the same compiler. I suggest you remove all build directories and compile asagi, easi, SeisSol again.

**Reply by buzhiqiuyuebai** [reply]
Thanks your suggestion!

**Reply by Thomas-Ulrich** [reply]
You should try spack to install SeisSol dependencies.
This would prevent such problems.
https://seissol.readthedocs.io/en/latest/installing-dependencies.html#spack-installation
